garden_ai/app/create.py

```diff
@@ -0,0 +1,172 @@
+import logging
+from datetime import datetime
+from pathlib import Path
+from typing import List, Optional
+
+import rich
+import typer
+from garden_ai.client import GardenClient
+from rich.prompt import Prompt
+
+logger = logging.getLogger()
+
+LOCAL_STORAGE = Path("~/.garden/db/").expanduser()
+(LOCAL_STORAGE / "gardens").mkdir(parents=True, exist_ok=True)
+(LOCAL_STORAGE / "pipelines").mkdir(parents=True, exist_ok=True)
+
+
+def setup_directory(directory: Optional[Path]) -> Optional[Path]:
+    """
+    Validate the directory provided by the user, scaffolding with "pipelines/" and
+    "models/" subdirectories if possible (i.e. directory does not yet exist or
+    exists but is empty).
+    """
+    if directory is None:
+        return None
+
+    if directory.exists():
+        if list(directory.iterdir()):
```

Comment:
```suggestion
        if len(list(directory.iterdir())) > 0:
```

Reply:
hmm yeah `list(...)` isn't the clearest truth condition, but I think the way I'd want to clean it up would be to replace it with something that plays nicely with iterators like `if any(directory.iterdir()): ...` rather than coercing to a list for the sake of `len`. Would also have the benefit of failing immediately in a nonempty directory rather than failing after `list`ing n directory items

Reply:
thanks for the attention to detail

Reply:
This "[truthy](https://www.freecodecamp.org/news/truthy-and-falsy-values-in-python/)" test is more pythonic. So, I'd say keep it as is